tests/integration/test_repos_release.py

```diff
@@ -151,17 +158,19 @@ def test_edit(self):
         """Test the ability to edit an existing asset."""
         self.basic_login()
         cassette_name = self.cassette_name('edit')
-        with self.recorder.use_cassette(cassette_name,
-                                        preserve_exact_body_bytes=True):
-            repository = self.gh.repository('github3py', 'github3.py')
+        with self.recorder.use_cassette(cassette_name):
+            repository = self.gh.repository('itsmemattchung', 'github3.py')
             release = repository.create_release(
                 '0.8.0.pre', 'develop', '0.8.0 fake release with upload',
                 'To be deleted'
                 )
-            with open(__file__) as fd:
-                asset = release.upload_asset(
-                    'text/plain', 'test_repos_release.py', fd.read()
-                    )
+        with self.recorder.use_cassette(cassette_name,
```

Comment:
So there's nothing saying that there can only be one cassette file per test. I know that's how other tests work here, but there's no rule about that. ;)

Just use two cassettes with different names and different settings for matching and then don't worry about having to change the record mode. :-D